Upgrade to Node 18-compliant dependencies

NSeydoux opened the pull request:
This adds support for Node 18 and Node 20. It fixes a couple of issues with the unit tests, but there are likely additional issues with the unchanged tests. A global review will be needed, but that's out of scope for this.

NSeydoux committed:
Upgrade to Node 18-compliant dependencies

NSeydoux commented:
As a note, mocking out `@inrupt/universal-fetch` resolves some of the issues in `src/components/text/index.test.tsx`. `undici` erroring out in `jsdom` is suprising to me, as it isn't something I've seen in out other libraries.

jeswr committed:
chore: use latest polyfills

jeswr commented:
We were on a version of our typescript-tooling libraries where the fetch polyfills didn't exist :)

jeswr committed:
feat: add node 20 support

NSeydoux committed:
Fix up mocks in the Text component

This fixes up the mocks in the Text component. In particular, it:
- removes side-effects between test by replacing shared objects by shared constructors
- replaces jest.spyOn mocks by actually setting expected values for the dataset, so that tests are more centered on behavior, and not implementation
- properly mocks solid-client, so that client-server operations are completely mocked. Using jest.spyOn still calls the underlying implementation, resulting in the fetch being called in this context, which isn't appropriate.

NSeydoux committed:
Avoid side effects in boolean component tests

Use shared constructors instead of shared objects

NSeydoux committed:
Fix mocks in boolean component tests

- Test for behavior rather than implementation by defining proper values in tests over using jest.spyOn
- Delete unnecessary references

NSeydoux committed:
Clear mocks before running tests instead of after
